### PR TITLE
Make returning execution data via signed URL optional

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Signed URLs in get execution data calls are deprecated, like [here](https://github.com/flyteorg/flyteidl/blob/6eff9b0be4f41b717b2d89697c7187a5216a0547/protos/flyteidl/admin/execution.proto#L307,L313). This PR introduces a config option to disable computing them for get execution data responses and defaults to omitting them altogether.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/1796

## Follow-up issue
_NA_

### CODE REVIEW

**Review by pmahindrakar-oss — DISMISSED** [review]

File: script/integration/k8s/integration.yaml [hunk 1]
```diff
@@ -296,6 +296,7 @@ data:
       region: "my-region"
       scheme: local
       signedUrls:
+        enabled: true
```
Comment (pmahindrakar-oss) [inline script/integration/k8s/integration.yaml:299]: If the default behavior is going to be false as per this https://github.com/flyteorg/flyteadmin/pull/283/files#diff-4149c9b64083f7b9e6680f7294d71f6ba58bc8da28a1189309eeea69d8ed07f3R63

Should we instead have integration test for false since that will be run without any config changes.
If possible we can have both the vairations in integration tests

Comment (katrogan) [reply script/integration/k8s/integration.yaml]: since changing configs in integration tests is a bit complicated I'll just remove this and adjust the tests accordingly to use the default behavior.

**Comment by katrogan** [comment]
PTAL @pmahindrakar-oss

**Review by EngHabu — DISMISSED** [review]
One more +1... because why not

**Review by pmahindrakar-oss — APPROVED** [review]

File: pkg/runtime/interfaces/application_configuration.go [hunk 2]
```diff
@@ -295,6 +295,9 @@ func (s *SchedulerConfig) GetReconnectDelaySeconds() int {
 
 // Configuration specific to setting up signed urls.
 type SignedURL struct {
+	// Whether signed urls should even be returned with GetExecutionData, GetNodeExecutionData and GetTaskExecutionData
+	// response objects.
+	Enabled bool `json:"enabled"`
```
Comment (EngHabu) [inline pkg/runtime/interfaces/application_configuration.go:300]:
```suggestion
	Enabled bool `json:"enabled" pflag:",Whether signed urls should even be returned with GetExecutionData, GetNodeExecutionData and GetTaskExecutionData response objects."`
```

**Review by EngHabu — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]
